### PR TITLE
Output format documentation

### LINKED ISSUE #2403: Support YAML OpenAPI/Swagger v2 definition generation
## 🚀 Feature

Currently the `protoc-gen-openapiv2` can output a JSON Swagger/OpenAPI v2 API definition file. It would be nice to have an option to produce a YAML version instead.

One main benefits of YAML over JSON is the support for comments. I originally came here with the intent to suggest a feature of generator plugin to add a `Code generated by protoc-gen-openapiv2. DO NOT EDIT.` header comment to the generated Swagger definition (similar to other protoc compiler plugins). However, I realize that is not possible with JSON.

I also personally find YAML a bit more readable compared to JSON, but that is just personal preference.

### PR DESCRIPTION
**Pull request by hedhyw**
#### References to other Issues or PRs

Resolve #2403
Code implemented in #2579 

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?

Yes.

#### Brief description of what is fixed or changed

Added documentation about configuring output format: yaml/json.

#### Other comments

No.

### CODE REVIEW

File: docs/docs/mapping/customizing_openapi_output.md [hunk 1]
```diff
@@ -404,4 +404,190 @@ For a more in depth example see [visibility_rule_echo_service.proto](https://git
 - [`visibility_restriction_selectors=INTERNAL,visibility_restriction_selectors=PREVIEW`](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/visibility_rule_preview_and_internal_echo_service.swagger.json)
 - [Not set](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/visibility_rule_none_echo_service.swagger.json)
 
+### Output format
+
+By default output format is JSON, but it is possible to configure it. For this, there is an option `output_format`. Allowed values are: `json`, `yaml`. The output format will also change an extension of the output files.
```
Comment (johanbrandhorst) [inline docs/docs/mapping/customizing_openapi_output.md:409]:
```suggestion
By default the output format is JSON, but it is possible to configure it using the `output_format` option. Allowed values are: `json`, `yaml`. The output format will also change the extension of the output files.
```

File: docs/docs/mapping/customizing_openapi_output.md [hunk 2]
```diff
@@ -404,4 +404,190 @@ For a more in depth example see [visibility_rule_echo_service.proto](https://git
...
+
+Example definition in the `buf.gen.yaml`:
+```yaml
+  - name: openapiv2
+    out: pkg
+    opt: output_format=yaml
+```
+
+Input example:
+```proto3
```
Comment (johanbrandhorst) [inline docs/docs/mapping/customizing_openapi_output.md:419]: Use `protobuf` for highlighting

File: docs/docs/mapping/customizing_openapi_output.md [hunk 3]
```diff
@@ -404,4 +404,190 @@ For a more in depth example see [visibility_rule_echo_service.proto](https://git
...
+message HelloReq {
+    string name = 1;
+}
+
+message HelloResp {
+    string message = 1;
+}
+```
+
+Output yaml (`opt: output_format=yaml`):
```
Comment (johanbrandhorst) [inline docs/docs/mapping/customizing_openapi_output.md:444]:
```suggestion
Output:
```

File: docs/docs/mapping/customizing_openapi_output.md [hunk 4]
```diff
@@ -404,4 +404,190 @@ For a more in depth example see [visibility_rule_echo_service.proto](https://git
 - [`visibility_restriction_selectors=INTERNAL,visibility_restriction_selectors=PREVIEW`](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/visibility_rule_preview_and_internal_echo_service.swagger.json)
 - [Not set](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/visibility_rule_none_echo_service.swagger.json)
 
+### Output format
+
+By default output format is JSON, but it is possible to configure it. For this, there is an option `output_format`. Allowed values are: `json`, `yaml`. The output format will also change an extension of the output files.
+
+Example definition in the `buf.gen.yaml`:
```
Comment (johanbrandhorst) [inline docs/docs/mapping/customizing_openapi_output.md:411]:
```suggestion
For example, if using `buf`:
```

File: docs/docs/mapping/customizing_openapi_output.md [hunk 5]
```diff
@@ -404,4 +404,190 @@ For a more in depth example see [visibility_rule_echo_service.proto](https://git
...
+  v1HelloResp:
+    type: object
+    properties:
+      message:
+        type: string
+```
+
+Output json (`opt: output_format=json`):
+```json
+{
```
Comment (johanbrandhorst) [inline docs/docs/mapping/customizing_openapi_output.md:504]: I don't think we need the JSON example, feel free to remove it.

**Review by johanbrandhorst — COMMENTED** [review]
Thank you for this!

File: docs/docs/mapping/customizing_openapi_output.md (hunk 2 above)
Comment (hedhyw) [reply docs/docs/mapping/customizing_openapi_output.md]: Fixed

File: docs/docs/mapping/customizing_openapi_output.md (hunk 3 above)
Comment (hedhyw) [reply docs/docs/mapping/customizing_openapi_output.md]: Fixed

File: docs/docs/mapping/customizing_openapi_output.md (hunk 5 above)
Comment (hedhyw) [reply docs/docs/mapping/customizing_openapi_output.md]: Removed

**Review by johanbrandhorst — APPROVED** [review]
LGTM

**Comment by johanbrandhorst** [comment]
Thank you for your contribution!

**Pull request merged**